File: app/src/main/java/s15666/pjwstk/pamobmi/ui/calories/MetricHBEquation.java
package s15666.pjwstk.pamobmi.ui.calories;

final class MetricHBEquation extends HarrisBenedictEquation {

    @Override
    double getGenderBaseValue(Gender gender) {
        return gender.equals(Gender.MALE) ?  66.5 : 655.1;
    }

    @Override
    double getWeightMultiplier(Gender gender) {
        return gender.equals(Gender.MALE) ? 13.75 : 9.56;
    }

    @Override
    double getHeightMultiplier(Gender gender) {
        return gender.equals(Gender.MALE) ? 5 : 1.85;
    }

    @Override
    double getAgeMultiplier(Gender gender) {
        return gender.equals(Gender.MALE) ? 6.75 : 4.67;
    }
}

File: app/src/main/java/s15666/pjwstk/pamobmi/benedictharris/MetricHBEquation.kt
package s15666.pjwstk.pamobmi.benedictharris

class MetricHBEquation : HarrisBenedictEquation() {

    override fun getGenderBaseValue(gender: Gender?): Double {
        return if (gender == Gender.MALE) 66.5 else 655.1
    }

    override fun getWeightMultiplier(gender: Gender?): Double {
        return if (gender == Gender.MALE) 13.75 else 9.56
    }

    override fun getHeightMultiplier(gender: Gender?): Double {
        return if (gender == Gender.MALE) 5.0 else 1.85
    }

    override fun getAgeMultiplier(gender: Gender?): Double {
        return if (gender == Gender.MALE) 6.75 else 4.67
    }
}